pkg/costmodel/costmodel.go

```diff
@@ -979,13 +982,121 @@ func (cm *CostModel) GetPricingSourceCounts() (*costAnalyzerCloud.PricingMatchMe
 	}
 }
 
+func getMachineTypeFromMap(machineType string, machineTypeMemoryMap map[string]int64) (int64, bool) {
+	memory, exists := machineTypeMemoryMap[machineType]
+	if !exists {
+		return 0, exists
+	}
+	return memory, exists
+}
+
+func getMachineType(projectID, zone, machineType string, ctx context.Context) (int64, error) {
+	computeService, err := compute.NewService(ctx)
+	if err != nil {
+		log.Warnf("Failed to list machine types: %v", err)
+		return 0, err
+	}
+	machineTypesListCall := computeService.MachineTypes.Get(projectID, zone, machineType)
+	machineTypes, err := machineTypesListCall.Do()
+	if err != nil {
+		log.Warnf("Failed to list machine types: %v", err)
+		return 0, err
+	}
+	return machineTypes.MemoryMb, nil
+
+}
+
+func GetMemorySize(projectID, zone, machineType string, machineTypeMemoryMap map[string]int64) (float64, error) {
+	ctx := context.Background()
+	
+	var exists bool
+	var err error
+	var memoryMb int64
+
+	if strings.Contains(machineType, "custom") {
+
+		str := (strings.Split(machineType, "-"))
+		memoryStr := str[len(str)-1]
+		memoryMb, err = strconv.ParseInt(memoryStr, 10, 64)
+		if err != nil {
+			log.Warnf("Error getting memory size from machine type string : %s", err.Error())
+
+			memoryMb, err = getMachineType(projectID, zone, machineType, ctx)
+			if err != nil {
+				log.Warnf("Failed to get machine type: %s  err: %s",machineType, err.Error())
+				return 0, err
+			}
+		}
+		
+	} else {
+		memoryMb, exists = getMachineTypeFromMap(machineType, machineTypeMemoryMap)
+
+		if !exists {
+
+			memoryMb, err = getMachineType(projectID, zone, machineType, ctx)
+
```

Comment:
If machineType doesn't exist in the map, its better to update it to avoid repeated calls to compute service.
```suggestion
machineTypeMemoryMap[machineType] = memoryMb
```